src/gameboard.js

```diff
@@ -17,27 +17,29 @@ export default class Gameboard {
   }
 
   receiveAttack(posX, posY) {
-    let coords = posToCoords(posX, posY);
-    if (this.board[coords]) {
-      if (this.board[coords].hasShip && this.board[coords].isHit === false) {
-        this.board[coords].isHit = true;
-        this.board[coords].hasShip.hit();
-        return true;
-      }
-    } else if (!this.board[coords]) {
-      return;
+    const coords = posToCoords(posX, posY);
+    if (!this.board[coords]) {
+      return false;
+    }
+    if (
+      this.board[coords] &&
```

Comment:
In loc sa verifici `this.board[coords]` si dupa `this.board[coords].isHit` poti folosi:
`this.board[coords]?.hasShip`. Cu semnul intrebarii, o sa dea `false` daca e `undefined`

Toata conditia ar putea fi `this.board[coords]?.hasShip && !this.board[coords].isHit`